fili-core/src/test/groovy/com/yahoo/bard/webservice/metadata/DataSourceMetadataLoadTaskSpec.groovy

```diff
@@ -24,46 +21,36 @@ import com.yahoo.bard.webservice.table.ConstrainedTable
 import com.yahoo.bard.webservice.table.StrictPhysicalTable
 import com.yahoo.bard.webservice.table.PhysicalTableDictionary
 
-import org.joda.time.DateTime
-import org.joda.time.DateTimeZone
 import org.joda.time.Interval
-import org.joda.time.format.DateTimeFormat
 
-import spock.lang.Shared
-import spock.lang.Specification
+class DataSourceMetadataLoadTaskSpec extends BaseDataSourceMetadataSpec {
 
-class DataSourceMetadataLoadTaskSpec extends Specification {
     private static final ObjectMappersSuite MAPPERS = new ObjectMappersSuite()
 
-    String tableName = TestDruidTableName.ALL_PETS.asName()
-
-    Interval interval1 = Interval.parse("2015-01-01T00:00:00.000Z/2015-01-02T00:00:00.000Z")
-    Interval interval2 = Interval.parse("2015-01-02T00:00:00.000Z/2015-01-03T00:00:00.000Z")
-    Interval interval3 = Interval.parse("2015-01-03T00:00:00.000Z/2015-01-04T00:00:00.000Z")
-    Interval interval123 = Interval.parse("2015-01-01T00:00:00.000Z/2015-01-04T00:00:00.000Z")
-
-    String version1 = DateTimeFormat.fullDateTime().print(DateTime.now().minusDays(1))
-    String version2 = DateTimeFormat.fullDateTime().print(DateTime.now())
-
-    TestApiDimensionName dim1 = TestApiDimensionName.BREED
-    TestApiDimensionName dim2 = TestApiDimensionName.SPECIES
-    TestApiDimensionName dim3 = TestApiDimensionName.SEX
+    @Override
+    def childSetupSpec() {
+        tableName = generateTableName()
+        intervals = generateIntervals()
+        dimensions = generateDimensions()
+        metrics = generateMetrics()
+        versions = generateVersions()
```

Comment:
We keep doing this in children classes. Since this simply assigns fields that are in the parent class, why not move this into the parent class?

Reply:
@archolewa Would you like to move `versions = generateVersions()` or all other `generate*()`'s ?

Reply:
I meant the entire method, sorry I wasn't clear.

Reply:
Well, If we move all of these and (in some cases) we override `generateIntervals()` in child Spec, for example, then other `generate` method might throw an error. For instance

```java
    @Override
    Map<String, Interval> generateIntervals() {
        [
                "interval1": Interval.parse(intervalString1),
                "interval3": Interval.parse(intervalString3),
                "interval13": Interval.parse(intervalString13),
                "interval23": Interval.parse(intervalString23)
        ]
    }
```

Then this default in base Spec won't work, because `intervals.interval2` does not exist

```java
    Map<String, DataSegment> generateSegments() {
        NumberedShardSpec partition1 = Stub(NumberedShardSpec) {
            getPartitionNum() >> 0
        }

        NumberedShardSpec partition2 = Stub(NumberedShardSpec) {
            getPartitionNum() >> 1
        }

        return [
                segment1: new DataSegment(
                        tableName,
                        intervals.interval1,
                        versions.version1,
                        null,
                        dimensions.keySet().toList(),
                        metrics.keySet().toList(),
                        partition1,
                        binaryVersions.binaryVersion1,
                        sizes.size1
                ),
                segment2: new DataSegment(
                        tableName,
                        intervals.interval1,
                        versions.version2,
                        null,
                        dimensions.keySet().toList(),
                        metrics.keySet().toList(),
                        partition2,
                        binaryVersions.binaryVersion1,
                        sizes.size2
                ),
                segment3: new DataSegment(
                        tableName,
                        intervals.interval2,
                        versions.version1,
                        null,
                        dimensions.keySet().toList(),
                        metrics.keySet().toList(),
                        partition1,
                        binaryVersions.binaryVersion1,
                        sizes.size1
                ),
                segment4: new DataSegment(
                        tableName,
                        intervals.interval2,
                        versions.version2,
                        null,
                        dimensions.keySet().toList(),
                        metrics.keySet().toList(),
                        partition2,
                        binaryVersions.binaryVersion1,
                        sizes.size2
                )
        ]
    }
```